src/hotspot/share/c1/c1_Optimizer.cpp

```diff
@@ -1198,6 +1201,28 @@ void NullCheckEliminator::handle_ProfileReturnType(ProfileReturnType* x) {
   x->set_needs_null_check(!set_contains(x->ret()));
 }
 
+void NullCheckEliminator::handle_Constant(Constant *x) {
+  ObjectType* ot = x->type()->as_ObjectType();
+  if (ot && ot->is_loaded()) {
+    ObjectConstant* oc = ot->as_ObjectConstant();
+    if (!oc || !oc->value()->is_null_object()) {
```

Comment:
Ditto, check `oc == nullptr`.

Now, the fact that `as_ObjectConstant` returns `nullptr` means this is not an _object constant_, but some other constant, right? I think this is similar to what other places in C1 do, so while awkward, this looks okay.

Reply:
fixed.